pkg/team/gcp.go

```diff
@@ -91,7 +93,7 @@ func (c Client) updateRoleBindingIfExists(bindings []*iamv1.Binding, role, names
 }
 
 func (c Client) createSecret(ctx context.Context, slug, teamID string) (*secretmanagerpb.Secret, error) {
-	return gcp.CreateSecret(ctx, c.gcpProject, c.gcpRegion, teamID, map[string]string{"team": slug})
+	return gcp.CreateSecret(ctx, c.gcpProject, c.gcpRegion, airflowGCPSecretPrefix+teamID, map[string]string{"team": slug})
```

Comment:
I'm not sure whether we should rename the existing team secret or if we should create a separate one for the airflow secrets backend feature. This secret might also be used for jupyterhubs, so to prefix it with airflow-variable might be confusing. Also, as `createSecret()` is only called on team create we would have to manually rename the existing team secrets if they want to opt in for the gsm secrets backend, causing the users having to update their code referencing the secret as well.

I'm thinking a better approach would be to create this secret on the airflow install event, as it is only needed for teams who are using airflow. If so, we could check on update airflow whether the secret exists and create it if not. That way we would only have to resync all airflows after this feature is merged and that will take care of the missing secrets for existing teams already using airflow.